external/llvm-project/mlir/include/mlir/Dialect/Arith/Utils/Utils.h

```diff
@@ -130,6 +130,10 @@ namespace arith {
 Value createProduct(OpBuilder &builder, Location loc, ArrayRef<Value> values);
 Value createProduct(OpBuilder &builder, Location loc, ArrayRef<Value> values,
                     Type resultType);
+
+// Map strings to float types.
+std::optional<FloatType> parseFloatType(Builder &b, StringRef name);
```

Comment:
... if we're doing this, we could probably rig it to just take a `MLIRContext*`, at the cost of a somewhat uglier implementation